docs/typescript.md

```diff
@@ -47,48 +84,42 @@ const App = () => <Link href="#">Click me</Link>
 
 ### Passing Props
 
-You can type the props of your styled components.\
-Unfortunately, you will need to pass a second parameter with the tag name because TypeScript is unable to infer the tagname.
+You can type the props of your styled components.
 
 ```jsx
 import styled from 'react-emotion'
 
 type ImageProps = {
   src: string,
+  width: number;
 }
 
-const Image = styled<ImageProps, 'div'>('div')`
-  background: url(${props => props.src}) center center;
+const Image0 = styled('div')`
+  width: ${(props: ImageProps) => props.width};
```

Comment:
Why are you using `width: ${(props: ImageProps)` instead of typing the component with `styled<ImageProps>('div')`? This is a bit confusing for me, since I see the latter as a much clearer way to type your component.

Reply:
Because with updated typing, there're no type parameters Props for `styled`.

Reply:
In TS 2.9 (which is in RC), you can use template string generic, however, before it, you should give a type to function and let TS infer correct type for component.

Reply:
Why? Typing every prop function with `${(props: ImageProps)` is more work, right? The redundant `div` in `styled<ImageProps, 'div'>` is a bit frustrating, but it will not be necessary anymore as soon as TypeScript 2.9 is released.

Reply:
> what you mentioned occurs only when you use more than two function argument, and as far as I experienced, it's not that common

In my experience I have a lot of these functions, if you are building a component library you can easily have ten of those prop functions. But if this isn't relevant anymore with TS 2.9 I digress.

Reply:
Maybe my opinion is from my **object syntax** based experience. Even when I make a library, I rarely use more than two function arguments.

Reply:
And, thank you for your feedback!